### PR TITLE
B-19507 limit customer access via validation code (MERGE AFTER B-19506)

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

This PR contains the `B-19506` branch https://github.com/transcom/mymove/pull/12488#pullrequestreview-1997774962 so ensure it is merged prior to this.

---

In the near future, we will be getting some validation code(s) from DPS that a customer will need to input when beginning their move. This is to keep users from creating fake moves and only really allow authenticated and appropriate users to use MilMove. 

In order to make this happen, we need the user to input that validation code before doing anything else. So the logic implemented in this BL item will require the user to put that code in if they haven't set their profile up (AKA have an empty profile) and the feature flag is turned on.

This PR does the following:
- creates a migration file that creates a table called `application_parameters` where the `validation_code` will be stored
- creates `internal` endpoint, `POST /validation-code`
- updates the `getWorkflowRoutes` to 
- refactors the `ConusOrNot.jsx` component to be an updated functional component
- created `ValidationCode.jsx` component to handle the form and API call
- added & updated all relevant tests, cleaned up some code
- my main man @TevinAdams added some playwright test adjustments when the flag is `true`

### How to test

Couple of setup things you need to do first:
- `make db_dev_migrate` (this will create the `application_parameters` table
- You'll need to fill in a row with a value in `validation_code` and `id`, `updated_at` and `created_at` will auto populate
- In your `.envrc`, set `export FEATURE_FLAG_VALIDATION_CODE_REQUIRED=true`
- Now fire up your `client` and `server`

**For the happiest path**
1. Access MM as a customer
2. Login as a brand new customer
3. You should now see the validation code screen
4. Input the code you slapped in the `application_parameters` table
5. It should now let you progress through the profile set up

**For the saddest path**
1. Access MM as a customer
2. Login as a brand new customer
3. You should now see the validation code screen
4. Input anything other than the code you put in the database
5. You should see an error screen and not be able to progress
6. You can have as many attempts as you want when inputting the code (this was a requirement from Kipp)

## Screenshots
Screenshot 2024-04-12 at 11 07 01 AM

Mobile
Screenshot 2024-04-12 at 11 21 56 AM

Screenshot 2024-04-12 at 11 12 31 AM

Mobile
Screenshot 2024-04-12 at 11 21 38 AM

### CODE REVIEW

File: config/env/prd.app-client-tls.env [hunk 1]
```diff
@@ -31,3 +31,4 @@ FEATURE_FLAG_PPM=false
 FEATURE_FLAG_NTS=false
 FEATURE_FLAG_NTSR=false
 FEATURE_FLAG_CAC_VALIDATED_LOGIN=false
+FEATURE_FLAG_VALIDATION_CODE_REQUIRED=true
```
[the remark this answers is not in the document]
Comment (danieljordan-caci) [reply config/env/prd.app-client-tls.env:34]: How bout you just merge that sucka

File: migrations/app/schema/20240411201158_add_application_parameter_and_validation_code_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-- this table will be used to hold validation codes a customer will enter prior to beginning their move
+
+CREATE TABLE IF NOT EXISTS application_parameters (
+	id uuid PRIMARY KEY NOT NULL,
+	validation_code TEXT NOT NULL,
```
Comment (cameroncaci) [inline migrations/app/schema/20240411201158_add_application_parameter_and_validation_code_table.up.sql:5]: Recommend swapping validation_code from `TEXT` to `VARCHAR(20)`, as the ticket mentioned these are 20 character codes

File: config/env/prd.app-client-tls.env (hunk 1 above)
[the remark this answers is not in the document]
Comment (cameroncaci) [reply config/env/prd.app-client-tls.env]: Cause then I'd have to stop my review and rebuild my app ⌛

File: config/flipt/storage/development.features.yaml [hunk 3]
```diff
@@ -25,6 +33,14 @@ flags:
...
       key: mil-app
       value: true
+- key: validation_code_required
+  name: Validaton code feature flag
+  type: BOOLEAN_FLAG_TYPE
+  enabled: true
+  rollouts:
+  - segment:
+      key: mil-app
+      value: true
```
Comment (cameroncaci) [inline config/flipt/storage/development.features.yaml:43]: What's this doing here..

**Review by cameroncaci — DISMISSED** [review]
Per slack discussion

File: config/flipt/storage/development.features.yaml (hunk 3 above)
Comment (danieljordan-caci) [reply config/flipt/storage/development.features.yaml]: Should be gone now

File: migrations/app/schema/20240411201158_add_application_parameter_and_validation_code_table.up.sql (hunk 2 above)
Comment (cameroncaci) [reply migrations/app/schema/20240411201158_add_application_parameter_and_validation_code_table.up.sql:5]: Per discussion, the 20 character alphanumeric appears to not be set in stone. To remain flexible, for other reviewers I recommend leaving this as `TEXT` with no constraints.

**Review by traskowskycaci — APPROVED** [review]
Works as expected!

**Review by TevinAdams — APPROVED** [review]

**Pull request merged**